### DISCUSSION TITLE
Assumptions of GLMM

### DISCUSSION THREAD

**Discussion by tatiana-pashkova** [RePsychLing/SMLP2023#28] [Q&A] [answered]
We've been talking about the assumptions of linear mixed models with a numerical response, for example, that the residuals need to be normally distributed and heteroscedastic. My question is, are there any assumptions for generalized linear mixed models (e.g. with a binomial response)?

**Comment by palday** [comment, marked as answer]
> be normally distributed and heteroscedastic.

:cough: Homoscedastic. :wink: :cough: 


You posed your question in terms of mixed models, but the same question works equally well for classical / non-mixed regression models.

The core assumptions are:
- the residuals (more precisely: the conditional distribution) follow the assumed distribution. 
  - For linear models, this is the normal distribution with same variance for all values of the linear predictor (that latter bit is the homoskedasticity assumption). 
  - For generalized linear models, this is whatever distribution you chose -- the `family` argument in R, or the 4th positional argument in Julia. 
  - Note that this assumption is on the *residuals*. For GL(M)Ms, this means that you've run everything through your linear predictor and inverse link function to get your predicted values and then residuals.
  - Finally, this assumption hides another one: independence. Unless you're doing a distribution where your including some type of autocorrelation structure, then you're generally assuming that your measurements are independent. Mixed models allow you to relax some this independence assumption (that's exactly what the grouping variables do!). The major result of violating independence is that your standard errors will be off. In intuitive terms, violating independence means that you effectively have fewer observations (non-independent observations => observations share information => each observation contributes less additional information). Since formulae for errors, precision, etc. depend on the number of observations, your error bars will be wrong (by being too small).
- Your predictors don't have noise in their own measurement. We often ignore this one in practice -- for categorical variables and many experimental manipulations, we fully control the values that the predictors take so it's clear that they're measured without noise. For things like e.g. word frequency, life gets more interesting. While we can exactly measure word frequency in a particular corpus, we usually actually care about word frequency in a particular language (or perhaps speaker group), in which case the corpus-based measure is an estimate of the True Value. Often these small errors don't matter in practice, but they can be very important for predictors measured with lots of noise.
